content/en/guides/directory-structure/nuxt-config.md

```diff
@@ -225,6 +207,67 @@ See more on the [plugins property](/guides/configuration-glossary/configuration
 
 </base-alert>
 
+### runtimeConfig
+
+#### `publicRuntimeConfig`
+
+- should hold all env variables that are public as these will be exposed on the frontend. This could include a reference to your public URL for example.
+- is available using `$config` in both server and client.
+
+```js{}[nuxt.config.js]
+export default {
+  publicRuntimeConfig: {
+    baseURL: process.env.BASE_URL || 'https://nuxtjs.org'
+  }
+}
+```
+
+#### `privateRuntimeConfig`
+
+- should hold all env variables that are private and that should not be exposed on the frontend. This could include a reference to your API secret tokens for example.
+- is only available on server using same `$config` (it overrides publicRuntimeConfig)
+
+```js{}[nuxt.config.js]
+export default {
+  privateRuntimeConfig: {
+    apiSecret: process.env.API_SECRET
+  }
+}
+```
+
+#### **Using your config values:**
+
+You can then access these values anywhere by using the context in your pages, store, components and plugins by using `this.$config` or `context.$config`.
+
+```html{}[pages/index.vue]
+<script>
+  asyncData ({ $config: { baseURL } }) {
+    const posts = await fetch(`${baseURL}/posts`)
+      .then(res => res.json())
+  }
+</script>
+```
+
+Inside your templates you can access your runtimeConfigs directly using `$config.*`
```

Comment:
Maybe worth mentioning that regardless, if rendering `$config.token` it will be displayed in page (someone had this question in one issue)

Reply:
warning added .let me know if it needs further explanation